theano/scalar/basic.py

```diff
@@ -1302,10 +1277,13 @@ def grad(self, inputs, output_gradients):
 class BinaryBitOp(BinaryScalarOp):
     def output_types(self, *input_types):
         t0, t1 = input_types[0]
+        if t0 == bool and t1 == bool:
+            return [bool]
         for i in input_types[0]:
-            if i not in (int8, int16, int32, int64):
-                raise TypeError('input to a BitOp must have type int8,'
-                                ' int16, int32 or int64... not %s' % i)
+            if i not in discrete_types:
```

Comment:
If you mean to exclude mixing bool and ints here, maybe you should use `int_types + uint_types`